pkg/util/test/util.go

```diff
@@ -2002,6 +2024,40 @@ func ValidateTelemetryInstalled(pxImageList map[string]string, cluster *corev1.S
 	return nil
 }
 
+// validatePodTopologySpreadConstraints validates pod topology spread constraints
+func validatePodTopologySpreadConstraints(deployment *appsv1.Deployment, timeout, interval time.Duration) error {
+	s := scheme.Scheme
+	apiextensionsv1.AddToScheme(s)
+	monitoringv1.AddToScheme(s)
+	k8sClient, err := NewK8sClient(s)
+	if err != nil {
+		return err
+	}
+	t := func() (interface{}, bool, error) {
+		existingDeployment, err := appops.Instance().GetDeployment(deployment.Name, deployment.Namespace)
+		if err != nil {
+			return nil, true, fmt.Errorf("failed to get deployment %s/%s", deployment.Namespace, deployment.Name)
+		}
+		expectedConstraints, err := util.GetTopologySpreadConstraints(k8sClient, existingDeployment.Spec.Template.Labels)
```

Comment:
This could be outside the retry function. We don't need to calculate the spread constraints on every retry.

Reply:
we need to get template labels from deployment template to construct the constraints, deployment labels can be changed, so it's better to get the labels in each retry.